File: src/main/java/com/andrew121410/mc/ccminecraftbot/packets/handle/pos/OnServerEntityPositionPacket.java
package com.andrew121410.mc.ccminecraftbot.packets.handle.pos;

import com.andrew121410.mc.ccminecraftbot.CCBotMinecraft;
import com.andrew121410.mc.ccminecraftbot.packets.PacketHandler;
import com.andrew121410.mc.ccminecraftbot.player.CCPlayer;
import com.github.steveice10.mc.protocol.packet.ingame.server.entity.ServerEntityPositionPacket;

public class OnServerEntityPositionPacket extends PacketHandler<ServerEntityPositionPacket> {
    @Override
    public void handle(ServerEntityPositionPacket packet, CCBotMinecraft ccBotMinecraft) {
        CCPlayer ccPlayer = ccBotMinecraft.getPlayer();
        if (packet.getEntityId() == ccBotMinecraft.getPlayer().getEntityId()) {
            ccPlayer.getCurrentLocation().add(
                    packet.getMoveX() / (128 * 32),
                    packet.getMoveX() / (128 * 32),
                    packet.getMoveX() / (128 * 32),
                    0f,
                    0f);
        }
    }
}

File: src/main/kotlin/com/andrew121410/mc/ccminecraftbot/packets/handle/pos/OnServerEntityPositionPacket.kt
package com.andrew121410.mc.ccminecraftbot.packets.handle.pos

import com.andrew121410.mc.ccminecraftbot.CCBotMinecraft
import com.andrew121410.mc.ccminecraftbot.packets.PacketHandler
import com.github.steveice10.mc.protocol.packet.ingame.server.entity.ServerEntityPositionPacket

class OnServerEntityPositionPacket : PacketHandler<ServerEntityPositionPacket>() {
    override fun handle(packet: ServerEntityPositionPacket, ccBotMinecraft: CCBotMinecraft) {
        val ccPlayer = ccBotMinecraft.player!!
        if (packet.entityId == ccPlayer.entityId) {
            ccPlayer.currentLocation!!.add(
                packet.moveX / (128 * 32),
                packet.moveX / (128 * 32),
                packet.moveX / (128 * 32),
                0f,
                0f
            )
        }
    }
}